rust/src/lockfile.rs

```diff
@@ -118,20 +118,25 @@ struct LockfileConfigMetadata {
 }
 
 #[derive(Serialize, Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
 struct LockfileRepoMetadata {
     generated: DateTime<Utc>,
 }
 
 #[derive(Serialize, Deserialize, Debug)]
-#[serde(untagged)]
+#[serde(untagged, deny_unknown_fields)]
 enum LockedPackage {
     Evr {
         evr: String,
         digest: Option<String>,
+        #[serde(skip_serializing)]
```

Comment:
> Are there really no use cases for us reading in a lockfile and writing it back out again, possibly modified?

We do that in a way with FCOS cosa builds today, where we input the base lockfile and the overrides and get back a new "flattened" base lockfile, but we don't really have a need to retain this metadata in the generated lockfile there. Could be interesting to do that though, so e.g. it could show up in the build browser or whatever.

It's not as trivial as just removing `skip_serializing` because the generated lockfile is actually done from scratch from the final list of `DnfPackage`s marked for install, rather than modifying the existing object. (I.e. there is no code path in rpm-ostree right now where it just re-serializes a parsed `LockfileConfig`.)

So in short, not today but might be worth doing in the future. Right now, I'm mostly interested in getting https://github.com/coreos/fedora-coreos-config/pull/965 wired up. And there all we care about is managing the input files (specifically the overrides overlay lockfile).